### DISCUSSION TITLE
How to add a validator to the path parameter of a request?

### DISCUSSION THREAD

**Discussion by Qparrot** [0b01001001/spectree#126] [Q&A] [answered]
Hi,
Thank you for all the work made with SpecTree, it is pretty neat!
I have a route '/api/distinct/<string:column>' that gets the unique values available in the specified column of my database. "column" can have only 3 values. I would like that OpenApi shows this thank to an enum.I am using flask, Is it possible to use an enum with [URL variables] ("https://flask.palletsprojects.com/en/1.1.x/quickstart/#variable-rules) ?

I tried this without success:

```python3
class Enum(str, Enum):
    nom = "nom"
    prenom ="prenom"
    commune = "commune"

class Query(BaseModel):
    column: Enum = Enum.nom

@utils.route('/distinct/<string:column>', methods=["GET"])
@spectree_api.validate(query=Query, resp=Response("HTTP_403", HTTP_200=DistinctResp), tags=["Utils"])
def get_distinct(column):
    ....
```

This adds a new query parameter to the request description in the swagger file. In my case this is not what I was aiming for: the "column" variable should be a path parameter.

**Comment by kemingy** [comment, marked as answer]
You almost got it. As you mentioned, flask [URL variables](https://flask.palletsprojects.com/en/1.1.x/quickstart/#variable-rules) doesn't support Enum. But you can use Enum in the URL query.

```py
class Enum(str, Enum):
    nom = "nom"
    prenom ="prenom"
    commune = "commune"

class Query(BaseModel):
    column: Enum = Enum.nom

@utils.route('/distinct', methods=["GET"])
@spectree_api.validate(query=Query, resp=Response("HTTP_403", HTTP_200=DistinctResp), tags=["Utils"])
def get_distinct(column):
    ....
```

Try to query this service with `http :8000/distinct?column=nom`

**Reply by Qparrot** [reply]
Thank you kemingy! I do it this way.
